src/state-prototypes.js

```diff
@@ -256,6 +258,22 @@ function addNewCellToState(state, cellType = 'code', language = 'js') {
   return state
 }
 
+function newAppMessage(appMessageId, appMessageText, appMessageDetails, appMessageWhen) {
+  return {
+    id: appMessageId,
+    message: appMessageText,
+    details: appMessageDetails,
+    when: appMessageWhen,
+  }
+}
+
+function addAppMessageToState(state, appMessage) {
+  const nextAppMessageId = newCellID(state.appMessages)
+  state.appMessages
+    .push(newAppMessage(nextAppMessageId, appMessage.message, appMessage.details, appMessage.when))
+  return state
+}
+
```

Comment:
i think this function should live in a `notebook-reducer-utils.js` file rather than in the state prototypes

Reply:
or it could go under `updateAppMessages` directly in notebook-reducer (since we already have a few helper functions there)

Reply:
Yeah, `addNewCellToState` is also in this file, so I figured it should go here as well. Happy to change it if you think it belongs elsewhere.